crates/bitwarden/src/auth/password.rs

```diff
@@ -16,6 +24,33 @@ pub(super) fn satisfies_policy(
     true
 }
 
+/// Validate if the provided password matches the password hash stored in the client.
+pub(super) async fn validate_password(
+    client: &Client,
+    password: String,
+    password_hash: String,
+) -> Result<bool> {
+    let login_method = client
+        .login_method
+        .as_ref()
+        .ok_or(Error::NotAuthenticated)?;
+
+    if let LoginMethod::User(login_method) = login_method {
+        match login_method {
+            UserLoginMethod::Username { email, kdf, .. }
+            | UserLoginMethod::ApiKey { email, kdf, .. } => {
+                let hash =
+                    determine_password_hash(email, kdf, &password, HashPurpose::LocalAuthorization)
+                        .await?;
+
+                Ok(hash == password_hash)
+            }
+        }
+    } else {
+        Ok(false)
```

Comment:
Should we maybe throw an error here?

Reply:
I think throwing an error makes sense, we're already throwing NotAuthenticated if the login_method is None, seems reasonable to also throw if it's invalid.